### PR TITLE
Ferminet forward pass complete

### PR DESCRIPTION
**Pull request by shaipranesh2**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by shaipranesh2** [comment]
Added docstrings, and implemented forward pass of model using the layers which were merged in. Also, changed prepare_hartree fock function to correctly calculate the hf orbital values

File: deepchem/models/tests/test_ferminet.py [hunk 1]
```diff
@@ -30,11 +35,9 @@ def test_FerminetModel():
 
 
 @pytest.mark.dqc
-def test_prepare_hf_solution():
-    # Test for the prepare_hf_solution function of FerminetModel class
-    H2_molecule = [['H', [0, 0, 0]], ['H', [0, 0, 0.748]]]
```
Comment (rbharath) [inline deepchem/models/tests/test_ferminet.py:35]: Why are we removing this function?

File: deepchem/models/tests/test_ferminet.py [hunk 2]
```diff
@@ -12,8 +12,13 @@
 except ModuleNotFoundError:
     pass
 
+try:
```
Comment (rbharath) [inline deepchem/models/tests/test_ferminet.py:15]: You shouldn't need this check (see the try/catch above)

File: deepchem/models/torch_models/ferminet.py [hunk 3]
```diff
@@ -26,48 +22,110 @@ def test_f(x: np.ndarray) -> np.ndarray:
 
 class Ferminet(torch.nn.Module):
     """Approximates the log probability of the wave function of a molecule system using DNNs.
```
Comment (rbharath) [inline deepchem/models/torch_models/ferminet.py:24]: Can you extend this docstring? Aim for a solid paragraph of docs and a usage example

File: deepchem/models/torch_models/ferminet.py [hunk 4]
```diff
@@ -181,41 +250,29 @@ def __init__(
...
             self.electron_no)  # initialize the position of the electrons
-        adam = optimizers.AdamW()
+        self.prepare_hf_solution()
         super(FerminetModel, self).__init__(
-            model, optimizer=adam,
-            loss=L2Loss())  # will update the loss in successive PR
+            self.model,
+            loss=torch.nn.MSELoss())  # will update the loss in successive PR
 
-    def prepare_hf_solution(self, x: np.ndarray) -> np.ndarray:
```
Comment (rbharath) [inline deepchem/models/torch_models/ferminet.py:206]: Why are we removing this argument?

**Review by rbharath — COMMENTED** [review]
@shaipranesh2 I have some questions about the implementation below

File: deepchem/models/tests/test_ferminet.py (hunk 1 above)
Comment (shaipranesh2) [reply deepchem/models/tests/test_ferminet.py:35]: I have split prepare hf function further into 2 smaller functions as I had to call only a part of the function repeatedly at each epoch(The function when given electron coordinates evaluates the hartree fock orbital values at those points. The initial few lines of the function initializes the hartree fock pyscf function with nuclear coordinates which can be done only once)

File: deepchem/models/torch_models/ferminet.py (hunk 4 above)
Comment (shaipranesh2) [reply deepchem/models/torch_models/ferminet.py:206]: I am splitting this function into 2 smaller functions. This function here initializes the hartree fock pyscf fucntion (which can be called only once at start), and the other parameters will be part of the function which will be added in the next PR (this function will be called whenever electrons are sampled).

File: deepchem/models/tests/test_ferminet.py (hunk 2 above)
Comment (shaipranesh2) [reply deepchem/models/tests/test_ferminet.py]: done

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**